src/Endpoint/VMware.ScriptRuntimeService.APIGateway/ScriptExecutionStorage/ScriptExecutionFileStorage.cs

```diff
@@ -160,19 +158,30 @@ public ScriptExecutionFileStorage(ILoggerFactory loggerFactory, ScriptExecutionS
          return reader.ReadScriptExecutionOutput();
       }
 
-      public INamedScriptExecution[] ListScriptExecutions(string userId) {
+      public INamedScriptExecution[] ListScriptExecutions(string userId, bool skipSystemExecutions) {
          var userScripts = ListScriptIds(userId);
 
          return userScripts.Select(scriptId => GetScriptExecution(userId, scriptId)).
-            Where(se => !string.IsNullOrEmpty(se?.Id)).
+            Where(se => 
+               !string.IsNullOrEmpty(se?.Id) &&
+               ShouldIncludeExecution(se, skipSystemExecutions)).
             ToArray();
       }
+
+      private static bool ShouldIncludeExecution(INamedScriptExecution execution, bool skipSystemExecutions) {
+         if (skipSystemExecutions && execution is NamedScriptExecution named) {
+            return !named.IsSystem;
```

Comment:
Why is `IsSytem` not in the `INamedScriptExecution`?

Reply:
I want the change to be contained in the API gateway only.